### ISSUE TITLE
Various Broken Symbols with Concrete Math OTF Font

### ISSUE THREAD

**Issue by CarrotDLaw** [typst/typst#5292]
### Description

1. Broken `sqrt` when the content inside gets tall enough. Seems being fixed in #5288 .
```typ
$ sqrt(sqrt(sqrt(sqrt(sqrt(sqrt(sqrt(sqrt(x)))))))) $
```
<img width="511" alt="image" src="https://github.com/user-attachments/assets/57b77ef8-447f-4e22-99fe-0620b97a3d79">

2. Superscripted `complement` makes brackets rendered in improperly large size. New Computer Modern has no such issue.
```typ
$ (A^c) = (A^sans(upright("c"))) = (A_complement) != (A^complement) $
```
<img width="463" alt="image" src="https://github.com/user-attachments/assets/a2ae108f-7c09-44d6-bcfb-9b676b477c06">

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by mkorje** [comment]
Regarding 1, as mentioned in the linked PR, this is a font issue. I've gotten in contact with the author of the font, and they have released a new version (0.64) which fixes this! It is already on [CTAN](https://ctan.org/pkg/concmath-otf?lang=en).

The behaviour in 2 isn't a bug. It's just that the `complement` symbol in Concrete Math is a fair bit taller than the one in New Computer Modern. This causes the overall height of the things inside the parentheses to be too tall for the default size parenthesis, and so it is replaced with a larger variant. You can always use the `math.lr` function with the `size` parameter to force the parentheses to be whatever size you like. For example, `$ lr((A^complement), size: #80%) $`.

**Comment by CarrotDLaw** [comment]
Thank you so much for your help! As 1 has been fixed and 2 is in fact not a bug, I'm now closing the issue. Though, I leave here a little workaround for those like me who wants to get rid of the surprisingly tall `complement` symbol in Concrete Math font.

```typ
#show math.equation: text.with(font: "Concrete Math")
#let complement = text(font: "New Computer Modern Math", str.from-unicode(0x2201))
$
  (A) (A^c) (A^sans(upright(c))) (A^complement) (A_complement) (A_sans(upright(c))) (A_c)
$
```

<img width="533" alt="image" src="https://github.com/user-attachments/assets/33d13439-6c5c-43fe-b0f3-f11cc272f9d3">

Using the New Computer Modern Math font for only the `complement` symbol lets typst to render brackets in preferrable size.
